### ISSUE TITLE
new Systemd commit doesn't start the server or something, before the commit works

### ISSUE THREAD

**Issue by durankeeley** [MysticRyuujin/guac-install#196]
Was having a problem with RDP telling me it couldn't log in, reverted back the script before systemd and confirm was able to log in first try on rdp.

Tested: Debian 10.10

PS: Debian 11 Bullseye works if you add the pretty name for LIBPNG="libpng-dev"

**Comment by MysticRyuujin** [comment]
hmmmm I'm not sure, haven't tested but these lines are in the script:
```
service guacd start
systemctl enable guacd
```

Do you have a log file from the run with systemd?

**Comment by durankeeley** [comment]
No error on "service guacd start"

![image](https://user-images.githubusercontent.com/5219040/131575853-6850e522-20c9-4600-b0c2-6323218e1b91.png)

same result after restart.

Killing the process: killall guacd

Then manually running "/usr/local/sbin/guacd -f" as root does then work.

I then killed the process and ran systemctl start guacd 

Then tried to connect while checking the status with systemctl status guacd....

The same connection that worked when I ran manually gets the following error:

RDP server closed/refused connection: Security negotiation failed (wrong security type?)
Sep 01 09:34:36 test-remotedesktop guacd[751]: guacd[751]: INFO:        RDP server closed/refused connection: Security negotiation failed (wrong security type?)

Looking into the /etc/systemd/system/guacd.service

It looks like the service runs as daemon

That took me to this site:
https://kifarunix.com/install-apache-guacamole-on-rocky-linux-8/

Basically had to make a user for guacd and set that us the run user.

**Comment by MysticRyuujin** [comment]
So you're saying that the systemd file runs guacd as a daemon, and it doesn't work, but if you change systemd to run as a user, it works fine?

What if it runs as the tomcat user?

I wonder if there's a way to either a) not run as daemon, or b) specify a user and just run as tomcat haha

**Comment by durankeeley** [comment]
Tried tomcat and root as user, no go.... tried my user and that worked fine

sed -i 's/daemon/durankeeley/' /etc/systemd/system/guacd.service

**Comment by durankeeley** [comment]
Had to run:

systemctl daemon-reload 
to reload the file then

systemctl start guacd


It's like it doesn't like system accounts

**Comment by MysticRyuujin** [comment]
hmmmm I don't know how to handle this...as far as I can tell there is a user `daemon` and it should work fine with `root` too. I don't know if this is unique to your system/setup/os. I'll have to test more I guess.

**Comment by durankeeley** [comment]
tried multiple fresh vms of debian 10, debian 11 and ubuntu server 20.04 and all the same

**Comment by durankeeley** [comment]
I wonder if it's because it needs a home folder....

**Comment by durankeeley** [comment]
got it!

http://mail-archives.apache.org/mod_mbox/guacamole-user/202108.mbox/%3CBYAPR01MB44561E05796F7C077935B22DCDC99@BYAPR01MB4456.prod.exchangelabs.com%3E

"so I created /usr/sbin/.config/freerdp and changed to owner to daemon since guacd is run as
daemon. Certs got stored and RDP is working now!"

**Comment by durankeeley** [comment]
Confirmed on a new vm

sudo mkdir -p /usr/sbin/.config/freerdp
Ran the script.

worked first time

**Comment by MysticRyuujin** [comment]
Did you also have to change the owner of the folder? `sudo` would make it owned by root?

**Comment by durankeeley** [comment]
yip i did chown daemon:daemon /usr/sbin/.config/freerdp after creation of the folder

**Comment by MysticRyuujin** [comment]
You rock! Thank you!

**Comment by durankeeley** [comment]
Confirmed new commit works great!
